docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,237 @@ The remaining points can happen any time after 1) is complete although note that
+
+`cylc validate` - as per existing command with the following changes:
+* If validating a workflow installed with `cylc install` then use any relevant
+  command line options specified as part of the install unless overridden on the
+  command line.
+* Will work fine on source directories which include `rose-suite.conf` files.
+* No `--strict` mode
+  (naked dummy tasks no longer allowed - all tasks must be declared).
+  * This one is controversial and requires further discussion!
+* Cyclic graph validation can be quite slow and is currently included as part of
+  `--strict`. Add a `--no-cyclic-graph-validation` option to disable this?
+
+Several combinations of the above commands will be commonly used.
+Propose to support these as separate commands:
+* `cylc install-play` (`cylc ip` for short?)
+* `cylc install-validate`
+* `cylc reinstall-reload`
+
+`cylc pause|unpause` - new commands replacing `cylc hold|release|unhold`
+* Note that when you use `cylc stop` the scheduler will be modified to unpause
+  the workflow before stopping.
+  This means that a stopped workflow will not be paused when it is continued via
+  `cylc play`. On the other, a workflow which is continued for any other reason
+  (e.g. the server died) will retain its paused status.
+
+`cylc register` - remove: functionality no longer required.
+* `cylc install` will now perform the task of associating the source and run
+  directories via a symlink.
+* `cylc play` has no need to do this since the workflow definition will now
+  always be found in the run directory.
+
+`cylc clean` - new command to replace `rose suite-clean`.
+* This should support options which allow you to just clean subsets of the data.
+  For example, if you clean the `log` directory this allows you to rerun a
+  workflow (keeping the existing data in `share` if that is what you want).
+  The default will be to remove everything including the top level directory.
+* Support an option to clean all install targets other than localhost?
+* We may also want an option to archive the log directory rather than remove it.
+  This could simply imply renaming the directory, perhaps with a timestamp
+  (similar to what `rose suite-run` does but without tar gzipping the
+  directory). This would allow a workflow to be warm started without completely
+  losing the previous log files (this is an operational requirement at present
+  although we hope that re-flows will reduce or remove the need for this).
+  Note that we would not provide any further support for these archived log
+  directories (e.g. no support for accessing the logs via the UI).
+* The file `log/rose-suite-run.locs` is currently required by `rose suite-clean`
+  but `cylc clean` should probably get this information from the database?
+
+Note that many commands currently have REG / SUITE as a required argument
+(e.g. `cylc validate`). Where possible these will be changed to be optional
+arguments with the default being `$PWD`.
```

Comment:
Good, I've always hated that use of `REG` :+1:

Reply:
Note that we still have the use case of `cylc stop <name>` and the desired functionality of `cylc stop <pattern>`, `cylc clean <pattern>`, etc.

Reply:
(My point there was just that I never liked - although it was probably my fault! - use of the unintuitive "REG" metavar.  It should have been "SUITE-NAME" or similar.)